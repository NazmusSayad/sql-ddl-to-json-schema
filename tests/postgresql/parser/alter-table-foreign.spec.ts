import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table adding foreign key with two columns, match and trigger.': {
    queries: [
      `ALTER TABLE people add constraint xyz foreign key fk_ax_id__ay_id (ax_id (20) asc , ay_id ) references other ( xid ( 10 ) desc , yid ) match full on delete set null on update no action on delete cascade;`,
    ],
  },

  'Parser: Should alter table adding foreign key with two columns, and match.': {
    queries: [
      `ALTER TABLE people add constraint xyz foreign key fk_ax_id__ay_id(ax_id(20)asc,ay_id)references other(xid(10)desc,yid)match full;`,
    ],
  },

  'Parser: Should alter table adding foreign key with one column.': {
    queries: [
      `ALTER TABLE people add constraint xyz foreign key fk_o_id (o_id) references other (id);`,
    ],
  },

  'Parser: Should alter table adding foreign key with unnamed key.': {
    queries: [`ALTER TABLE people add constraint xyz foreign key (o_id) references other (id);`],
  },

  'Parser: Should alter table adding foreign key with unnamed constraint.': {
    queries: [
      `ALTER TABLE people add constraint foreign key fk_o_id (o_id) references other (id);`,
      `ALTER TABLE people add foreign key fk_o_id (o_id) references other (id);`,
    ],
  },

  'Parser: Should alter table adding foreign key with unnamed constraint and key.': {
    queries: [`ALTER TABLE people add foreign key (o_id) references other (id);`],
  },

  'Parser: Should alter table dropping foreign key.': {
    queries: [`ALTER TABLE people drop foreign key fk_oid;`],
  },
});
