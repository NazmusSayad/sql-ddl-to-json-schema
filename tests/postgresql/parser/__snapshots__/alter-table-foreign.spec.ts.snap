// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter table adding foreign key with one column. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "o_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": "xyz",
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "id",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": null,
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with two columns, and match. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "ax_id",
                            "length": 20,
                            "sort": "asc",
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                        {
                          "def": {
                            "column": "ay_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": "xyz",
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "xid",
                                "length": 10,
                                "sort": "desc",
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                            {
                              "def": {
                                "column": "yid",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": "full",
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with two columns, match and trigger. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "ax_id",
                            "length": 20,
                            "sort": "asc",
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                        {
                          "def": {
                            "column": "ay_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": "xyz",
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "xid",
                                "length": 10,
                                "sort": "desc",
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                            {
                              "def": {
                                "column": "yid",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": "full",
                          "on": [
                            {
                              "action": "set null",
                              "trigger": "delete",
                            },
                            {
                              "action": "no action",
                              "trigger": "update",
                            },
                            {
                              "action": "cascade",
                              "trigger": "delete",
                            },
                          ],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with unnamed constraint and key. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "o_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": null,
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "id",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": null,
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with unnamed constraint. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "o_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": null,
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "id",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": null,
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with unnamed constraint. 2`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "o_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": null,
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "id",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": null,
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table adding foreign key with unnamed key. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "addForeignKey",
                      "columns": [
                        {
                          "def": {
                            "column": "o_id",
                            "length": undefined,
                            "sort": undefined,
                          },
                          "id": "P_INDEX_COLUMN",
                        },
                      ],
                      "name": "xyz",
                      "reference": {
                        "def": {
                          "columns": [
                            {
                              "def": {
                                "column": "id",
                                "length": undefined,
                                "sort": undefined,
                              },
                              "id": "P_INDEX_COLUMN",
                            },
                          ],
                          "match": null,
                          "on": [],
                          "table": "other",
                        },
                        "id": "P_COLUMN_REFERENCE",
                      },
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;

exports[`Parser: Should alter table dropping foreign key. 1`] = `
ParsedQuerySerializer {
  "parsed": {
    "def": [
      {
        "def": {
          "def": {
            "specs": [
              {
                "def": {
                  "spec": {
                    "def": {
                      "action": "dropForeignKey",
                      "key": "fk_oid",
                    },
                    "id": "O_ALTER_TABLE_SPEC",
                  },
                },
                "id": "P_ALTER_TABLE_SPECS",
              },
            ],
            "table": "people",
          },
          "id": "P_ALTER_TABLE",
        },
        "id": "P_DDS",
      },
    ],
    "id": "MAIN",
  },
}
`;
